test/powershell/Modules/Microsoft.PowerShell.Core/CompatiblePSEditions.Module.Tests.ps1

```diff
@@ -327,6 +327,26 @@ Describe "Import-Module from CompatiblePSEditions-checked paths" -Tag "CI" {
                 [System.Management.Automation.Internal.InternalTestHooks]::SetTestHook("TestWindowsPowerShellVersionString", $null)
             }
         }
+
+        It "Current location in Windows PS mirrors local current location" -TestCases $failCases -Skip:(-not $IsWindows) {
+            param($Editions, $ModuleName, $Result)
+            $location = Join-Path $TestDrive "Custom dir" (New-Guid).ToString()
+            $null = New-Item -Path $location -ItemType Directory
+            Push-Location -Path $location
+
+            # right after module import remote $PWD should be synchronized
+            Import-Module $ModuleName -UseWindowsPowerShell
+            $s = Get-PSSession -Name WinPSCompatSession
+            (Invoke-Command -Session $s {Get-Location}).Path | Should -BeExactly $PWD.Path
+
+            # after local $PWD changes remote $PWD should be synchronized
+            Set-Location -Path ..
+            (Invoke-Command -Session $s {Get-Location}).Path | Should -BeExactly $PWD.Path
+
+            # after WinCompat cleanup local $PWD changes should not cause errors
+            Remove-module $ModuleName -Force
+            Pop-Location
```

Comment:
The Pop-Location should be in a try/finally to ensure we revert to the original setting.

Reply:
Updated.